src/test/java/org/opensearch/geospatial/index/query/GeospatialQueryTestHelperIT.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.query;
+
+import static org.opensearch.common.xcontent.ToXContent.EMPTY_PARAMS;
+import static org.opensearch.geospatial.GeospatialTestHelper.randomLowerCaseString;
+import static org.opensearch.index.query.AbstractGeometryQueryBuilder.DEFAULT_SHAPE_FIELD_NAME;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.common.geo.GeoJson;
+import org.opensearch.common.geo.ShapeRelation;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geospatial.GeospatialRestTestCase;
+import org.opensearch.geospatial.index.mapper.xyshape.XYShapeFieldMapper;
+import org.opensearch.geospatial.index.query.xyshape.XYShapeQueryBuilder;
+
+public class GeospatialQueryTestHelperIT extends GeospatialRestTestCase {
+    private static final String SHAPE_RELATION = "relation";
+    private static final String INDEXED_SHAPE_FIELD = "indexed_shape";
+    private static final String SHAPE_INDEX_FIELD = "index";
+    private static final String SHAPE_ID_FIELD = "id";
+    private static final String SHAPE_INDEX_PATH_FIELD = "path";
+
+    public void queryTestUsingNullShape(String indexName, String fieldName, String contentType) throws Exception {
+        createIndex(indexName, Settings.EMPTY, Map.of(fieldName, contentType));
+        String body = buildContentAsString(builder -> builder.field(fieldName, (String) null));
+        String docID = indexDocument(indexName, body);
+
+        final Map<String, Object> document = getDocument(docID, indexName);
+        assertTrue("failed to index document with type", document.containsKey(fieldName));
+        assertNull("failed to accept null value", document.get(fieldName));
+
+        deleteIndex(indexName);
+    }
```

Comment:
You can move helper methods inside GeospatialRestTestCase instead of creating a new composition

Reply:
if XYPoint and XYShape similar test cases, then we can create AbstractXYShapeQueryTestCases and move common tests there.

Reply:
refactored the tests